### PR TITLE
Add `withdraw` command and rename `register_service `

### LINKED ISSUE #743: Add withdraw command to register_service.py
We need a way to withdraw tokens that have been deposited by the service registry. This can't be easily done via Etherscan, since each deposit gets its own instance of the `Deposit` contract, so that those are not verified.

### PR DESCRIPTION
**Pull request by karlb**
Closes #743.

### CODE REVIEW

**Comment by karlb** [comment]
Do we have any documentation on that command that needs updating? We don't have any in this repo.

**Comment by palango** [comment]
> Do we have any documentation on that command that needs updating? We don't have any in this repo.

Doesn't seem like it, so far it was for internal use only. Wonder if we should add it here or to the RSB...

File: src/raiden_libs/cli.py [hunk 1]
```diff
@@ -187,7 +188,9 @@ def blockchain_options(contracts: List[str]) -> Callable:
         CONTRACT_MONITORING_SERVICE: "monitor-contract",
         CONTRACT_ONE_TO_N: "one-to-n-contract",
         CONTRACT_SERVICE_REGISTRY: "service-registry-contract",
+        CONTRACT_DEPOSIT: "service-deposit-contract",
```
Comment (palango) [inline src/raiden_libs/cli.py]: what about `service-registry-deposit-contract`?

File: src/raiden_libs/contract_info.py [hunk 2]
```diff
@@ -44,8 +44,9 @@ def get_contract_addresses_and_start_block(
 
     # Get deployed addresses for those contracts which have no overwrites
     addresses = {
-        c: address_overwrites.get(
-            c, to_canonical_address(contract_data["contracts"][c]["address"])
+        c: (
+            address_overwrites.get(c)
+            or to_canonical_address(contract_data["contracts"][c]["address"])
```
Comment (palango) [inline src/raiden_libs/contract_info.py]: Is there a difference here?

File: src/raiden_libs/service_registry.py [hunk 3]
```diff
@@ -180,5 +187,55 @@ def update_service_url(
         )
 
 
+@blockchain_options(contracts=[CONTRACT_DEPOSIT])
+@cli.command("withdraw")
+@click.option(
+    "--to", type=str, callback=validate_address, help="Target address for withdrawn tokens"
```
Comment (palango) [inline src/raiden_libs/service_registry.py]: Is this automatically required?

File: src/raiden_libs/service_registry.py [hunk 4]
```diff
@@ -180,5 +187,55 @@ def update_service_url(
...
+    deposit_contract = contracts[CONTRACT_DEPOSIT]
+
+    # Check usage of correct key
+    withdrawer = deposit_contract.functions.withdrawer().call()
+    caller_address = private_key_to_address(private_key)
+    if to_canonical_address(withdrawer) != caller_address:
+        log.error(
+            "You must used the key used to deposit when withdrawing",
+            expected=withdrawer,
+            actual=caller_address,
```
Comment (palango) [inline src/raiden_libs/service_registry.py]: This should be checksum encoded

File: src/raiden_libs/service_registry.py [hunk 5]
```diff
@@ -180,5 +187,55 @@ def update_service_url(
...
+        )
+        sys.exit(1)
+
+    # Can we withdraw already?
+    release_at = deposit_contract.functions.release_at().call()
+    if time.time() < release_at:
+        log.error("Too early to withdraw")
```
Comment (palango) [inline src/raiden_libs/service_registry.py]: Would be nice to log how much time is missing...

File: src/raiden_libs/service_registry.py [hunk 6]
```diff
@@ -180,5 +187,55 @@ def update_service_url(
...
+        sys.exit(1)
+
+    # Withdraw now!
+    log.info("Withdrawing", to=to_checksum_address(to))
+    deposit_contract.functions.withdraw(to).transact({"from": caller_address})
```
Comment (palango) [inline src/raiden_libs/service_registry.py]: Can you use `checked_transact` here?

File: deployment/docker-compose.yml [hunk 7]
```diff
@@ -213,7 +213,7 @@ services:
   registration-goerli:
     <<: *defaults
     restart: "no"
-    command: ["python3", "-m", "raiden_libs.register_service"]
+    command: ["python3", "-m", "raiden_libs.service_registry"]
```
Comment (palango) [inline deployment/docker-compose.yml]: Don't we need to add `register` here?

File: src/raiden_libs/cli.py (hunk 1 above)
Comment (karlb) [reply src/raiden_libs/cli.py]: I thought "user-deposit" and "service-deposit" were a nice pair, and it's a bit shorter. But I don't really mind either way.

File: src/raiden_libs/contract_info.py (hunk 2 above)
Comment (karlb) [reply src/raiden_libs/contract_info.py]: Yes, a very subtle one. In the old version the fallback was evaluated in every case, causing a key error for the deposit contract.

File: src/raiden_libs/cli.py (hunk 1 above)
Comment (palango) [reply src/raiden_libs/cli.py]: Fine for me as well. BTW, how do people get the address? Do they have to filtere the events themselves?

Can we log it during registration at least?

File: src/raiden_libs/service_registry.py (hunk 3 above)
Comment (karlb) [reply src/raiden_libs/service_registry.py]: No, it isn't. I'll change it. I first had it as an `argument`, which is always required. But then I noticed we don't use arguments anywhere and made it an option, again.

File: src/raiden_libs/service_registry.py (hunk 4 above)
Comment (karlb) [reply src/raiden_libs/service_registry.py]: Right, I lost that when introducing the `caller_address` variable.

File: deployment/docker-compose.yml (hunk 7 above)
Comment (karlb) [reply deployment/docker-compose.yml]: Yes, well spotted!

File: src/raiden_libs/service_registry.py (hunk 5 above)
Comment (karlb) [reply src/raiden_libs/service_registry.py]: Done.

File: src/raiden_libs/service_registry.py (hunk 6 above)
Comment (karlb) [reply src/raiden_libs/service_registry.py]: Done.

**Comment by karlb** [comment]
Great feedback! I've updated the code and also included one more cleanup (see last commit).

File: src/raiden_libs/cli.py (hunk 1 above)
Comment (karlb) [reply src/raiden_libs/cli.py]: I used Etherscan when testing. I wanted to get this quickly done, so I skipped event filtering and automated tests. Both make sense in later PRs.

**Pull request merged**